applications/OptimizationApplication/python_scripts/output/optimization_info_ascii_writer.py

```diff
@@ -0,0 +1,300 @@
+from datetime import datetime
+from abc import ABC
+from abc import abstractmethod
+from typing import Union
+
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.OptimizationApplication.optimization_info import OptimizationInfo
+from KratosMultiphysics.time_based_ascii_file_writer_utility import TimeBasedAsciiFileWriterUtility
+from KratosMultiphysics.OptimizationApplication.output.output_utilities import GetOptimizationInfoAvailableKeysForType
+from KratosMultiphysics.OptimizationApplication.output.output_utilities import GetPatternMatchedOptimizationKeys
+from KratosMultiphysics.OptimizationApplication.output.output_utilities import GetPrefixWithoutDataName
+from KratosMultiphysics.OptimizationApplication.output.output_utilities import GetPlaceHolderWithValues
+
+class OptimizationAsciiInfo(ABC):
```

Comment:
Restrictions:

There should be three categories to read from `ResponseData`, `ControlData`, `AlgorithmData`. The defaults list of strings should be provided for each category.


```json
            "optimization_info_output_settings": [
                "response_data_names": ["this_name"],
                "control_data_names": [],
                "algorithm_data_names": [],
               "precision": "0.4e"
            ]
```